### ISSUE TITLE
2.14.0 breaking change

### ISSUE THREAD

**Issue by h3rmanj** [getsentry/sentry-cli#1495]
### Environment

`getsentry/sentry-cli:2.14.0` on docker hub.

### Steps to Reproduce

Running these in my CI

```
    - sentry-cli releases new $VERSION
    - sentry-cli releases files $VERSION upload-sourcemaps --url-prefix '~/assets' build/assets --rewrite
    - sentry-cli releases finalize $VERSION
    - sentry-cli releases set-commits --auto $VERSION || echo 'Found no commits to assign to release.'
    - sentry-cli releases deploys $VERSION new -e $SENTRY_ENV
```

### Expected Result

Succeess

### Actual Result

`sentry-cli releases files $VERSION upload-sourcemaps --url-prefix '~/assets' build/assets --rewrite` command gives this error:  
`error: A release slug is required (provide with --release)`

### Workaround

Pinned my cli version to `2.13.0`

**Comment by kamilogorek** [comment]
cc @vaind it apparently broke with https://github.com/getsentry/sentry-cli/pull/1489 as we dont have integration tests for legacy APIs :(

@h3rmanj in the meantime you can fix this by replacing

```
sentry-cli releases files $VERSION upload-sourcemaps --url-prefix '~/assets' build/assets --rewrite
```

with

```
sentry-cli sourcemaps upload build/assets --release $VERSION --url-prefix '~/assets'
```

(`--rewrite` is a default in v2, so no need to set it)

**Comment by fdintino** [comment]
Which part is the legacy API? The [current sentry docs](https://docs.sentry.io/product/cli/releases/#finalizing-releases) have the release version passed as a positional argument, not with `--release`.

**Comment by fubhy** [comment]
This broke our github ci pipeline too.

**Comment by pgraemer** [comment]
same here.

**Comment by eXaminator** [comment]
I'm with @kamilogorek. This also broke our pipeline and we also did it as described in the docs. So I hope this will be fixed soon and if this API is legacy / deprecated it should be documented as such.

**Comment by kamilogorek** [comment]
re docs: https://github.com/getsentry/sentry-cli/issues/1429

This is a "legacy" API, which is supposed to be fully compatible so that we don't have to update the docs right away. This issue is a bug and will be fixed asap, no need to migrate to another API in v2 :)

**Comment by avg07** [comment]
> cc @vaind it apparently broke with #1489 as we dont have integration tests for legacy APIs :(
> 
> @h3rmanj in the meantime you can fix this by replacing
> 
> ```
> sentry-cli releases files $VERSION upload-sourcemaps --url-prefix '~/assets' build/assets --rewrite
> ```
> 
> with
> 
> ```
> sentry-cli sourcemaps upload build/assets --release $VERSION --url-prefix '~/assets'
> ```
> 
> (`--rewrite` is a default in v2, so no need to set it)

This solves the problem

**Comment by exodusanto** [comment]
Downloading the last version of cli on my local machine i saw a new options flag in deploy command:

```
> sentry-cli releases deploys --help
sentry-cli-releases-deploys
Manage deployments for Sentry releases.

USAGE:
    sentry-cli releases deploys [OPTIONS] <VERSION> <SUBCOMMAND>

ARGS:
    <VERSION>    The version of the release

OPTIONS:
        --auth-token <AUTH_TOKEN>    Use the given Sentry auth token.
    -h, --help                       Print help information
        --header <KEY:VALUE>         Custom headers that should be attached to all requests
                                     in key:value format.
        --log-level <LOG_LEVEL>      Set the log output verbosity. [possible values: trace, debug,
                                     info, warn, error]
    -o, --org <ORG>                  The organization slug
    -p, --project <PROJECT>          The project slug.
        --quiet                      Do not print any output while preserving correct exit code.
                                     This flag is currently implemented only for selected
                                     subcommands. [aliases: silent]
    -r, --release <RELEASE>          The release slug. <-- HERE

SUBCOMMANDS:
    help    Print this message or the help of the given subcommand(s)
    list    List all deployments of a release.
    new     Creates a new release deployment.
```
Others commands doesn't have that flag, and it match the error, never see it before.

**Comment by vaind** [comment]
The issue should be fixed in v2.14.1

**Comment by kamilogorek** [comment]
Can confirm. Regression fixed in `2.14.1`. New binaries and a Docker container have been released. Thanks for your patience and excuse us for the interruption.

**Comment by eriktoyra** [comment]
`2.14.1` solved it for us also. Pipelines are happy again. Will make sure to lock down our pipeline to install `2.14.1` for now to prevent uncontrolled installation from `latest`. Thanks for fixing this so fast @kamilogorek @vaind. 🙌🏻
